crates/rpc-types-engine/src/envelope.rs

```diff
@@ -270,6 +291,17 @@ impl OpNetworkPayloadEnvelope {
     }
 }
 
+/// Errors that can occur when encoding a payload envelope.
+#[derive(Debug, Clone, PartialEq, Eq, thiserror::Error)]
+pub enum PayloadEnvelopeEncodeError {
+    /// Wrong versions of the payload.
+    #[error("Wrong version of the payload")]
+    WrongVersion,
+    /// An error occured during snap encoding.
+    #[error(transparent)]
+    SnapEncoding(#[from] snap::Error),
+}
```

Comment:
doe we need this additional error if we already have

PayloadEnvelopeError

Reply:
Are you alright if I add a variant to that to say the snap encoding failed?

Reply:
As well as the wrong version actually too

Reply:
Basically - just need these variants in `PayloadEnvelopeError` to be able to use it